### ISSUE TITLE
Achievements!

### ISSUE THREAD

**Issue by FruitieX** [CIVIS-project/YouPower#47]
Challenges, actions and the line between these seems a bit confusing at the moment.

What we could do instead is a system of achievements. These would be different in that you don't explicitly need to "participate" or "join" an achievement in any way, instead you just automatically "unlock" them by completing the required conditions (such as finishing 10 actions of any kind).

**Comment by FruitieX** [comment]
Marking as important because this influences quite a bit of (as of yet unfinished!) code that deals with actions & challenges.

**Comment by FruitieX** [comment]
It has been agreed that we will implement achievements instead of challenges.

**Comment by sanja7s** [comment]
Having achievements instead of challenges is good. 

-- one more note however would be, if possible, to have some way to tell to the people when they are some % close to achieving some of "achievements" as that way we could still keep the _goal setting_ aspect in our app. 
-- perhaps a user could see what are the closest achievements on the way for him?
would that be feasible?

**Comment by FruitieX** [comment]
Yep, we should be able to tell _how close_ a user is to completing a certain achievement. We could then display these sorted according to percentage completed on eg. the app front page.

**Comment by sanja7s** [comment]
that sounds perfect. in that way, people do not even need to subscribe to "achievements" but will still in a way get subscribed.

**Comment by FruitieX** [comment]
Yes, they just get that achievement right away when they have completed some goal/goals.

**Comment by FruitieX** [comment]
Ideas:
- Complete 1, 3, 5, 10 actions
- Save 10, 20, 50, 75, 100, 150, 200, 300, 500, 1000 leaves worth of energy
- Participate in a community and save together! (100, 250, 500, 1000, 3000, 10000 leaves saved in a community)
- Review an action
- Create an action
- Have someone complete one of your actions
- Participate in a community discussion by posting a comment
- Be a beta tester
- Use the app daily for one week
- Unlock 25%, 50%, 75%, 90% of all achievements

etc...

**Comment by sanja7s** [comment]
- Be first of your friends (belong to a community first to) achieve something

**Comment by sanja7s** [comment]
Good ideas, let us discuss later today their feasibility and which ones are the best to have

**Comment by FruitieX** [comment]
#62 implemented the achievements framework.

## TODO
- Unit tests
- Implement all achievement ideas discussed above

**Comment by FruitieX** [comment]
Several achievements are still awaiting other completion of other features, such as households, energy data...

Approx. half of the above ideas are now implemented in the `master` branch.
